### DISCUSSION TITLE
Dead keys for auxiliary bindings

### DISCUSSION THREAD

**Discussion by PhirosWolf** [OpenTabletDriver/OpenTabletDriver#2488] [Ideas]
This is just a suggestion. But dead keys, which are keys that have no actions on their own but provide alternative bindings when combined with another key (for example ctrl, shift and alt are dead keys), could be a good addition imo.

It would allow the user to have more shortcuts (`a=(x+1)(n-x)` where a is the number of registrable actions, n the number of available keys and x the number of dead keys instead of `a=n`) and therefore, a more efficient workflow (a 140% increase in the number of registrable actions for a tablet with 10 keys with only 2 dead keys).

I believe there wasn't any similar suggestions in the discussions, unless i've searched the wrong terms.

**Comment by jamesbt365** [comment]
so you want the inclusion of the modifier keys?

**Reply by PhirosWolf** [reply]
Yes. (From the formulation of your sentence it sounds like it's already been suggested (or implemented), did I miss something?)

**Reply by jamesbt365** [reply]
![image](https://user-images.githubusercontent.com/22662856/201135503-6094a2fc-4cb5-4c34-97a3-9f52861b97cd.png)

**Reply by PhirosWolf** [reply]
This is not what I mean. I suggested dead keys at the driver level, not the keyboard level.
Basically, the driver detects that button 1 of the tablet has been pressed, then detects the button 4 being pressed while button 1 is not released, and from that it will look for the alternative shortcut set for button 4 instead of the shortcut set for button 4. You see what I mean?

**Comment by gonX** [comment]
I don't really think this is particularly easy to implement. It would require OTD also being able to read keyboard state, which I fear would put us in the same category as a keylogger.
